core/domain/feedback_services.py

```diff
@@ -720,6 +720,10 @@ def _get_all_recipient_ids(exploration_id, thread_id, author_id):
     batch_recipient_ids = owner_ids - sender_id
     other_recipient_ids = participant_ids - batch_recipient_ids - sender_id
 
+    # Remove anonymous users from recipient list.
```

Comment:
(That way, you know that any of these lists always consists of strings that are valid user_ids, and you'll never have to make a mental transition from "lists that may or may not contain None" to "lists whose elements are guaranteed to be fully valid".)